tests/test_os2mo.py

```diff
@@ -6,30 +6,130 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-
+import os
 import unittest
-from mox_stsorgsync import os2mo
+os.environ["STSORGSYNC_PHONE_SCOPE_CLASSES"] = " 1 2 "
+from mox_stsorgsync import os2mo  # noqa
+initial_settings = dict(os2mo.settings)
 
 
 class Tests(unittest.TestCase):
 
+    def test_env_vs_config(self):
+        self.assertEqual(
+            initial_settings["STSORGSYNC_PHONE_SCOPE_CLASSES"],
+            ["1", "2"]
+        )
+
     def test_os2mo_url(self):
         self.assertTrue(os2mo.os2mo_url("{BASE}").endswith("service"))
 
-    def test_addresses_to_user(self):
+    def test_addresses_to_user_dar(self):
         user = {}
         os2mo.addresses_to_user(user, [
             {'address_type': {'scope': "DAR"}, 'name': 'ABC', 'uuid': "XYZ"}
         ])
         self.assertEqual(user, {'Location': {'Uuid': 'XYZ', 'Value': 'ABC'}})
 
+    def test_secondary_address_when_first_is_secret(self):
+        "only the classes given are chosen and only of PUBLIC or no visibility"
+        user = {}
+        os2mo.settings["STSORGSYNC_PHONE_SCOPE_CLASSES"] = [
```

Comment:
Consider using `patch` for temporarily overriding settings.

Reply:
I shall!